apps/aecore/src/aec_block_micro_candidate.erl

```diff
@@ -101,22 +101,15 @@ int_create(BlockHash, Block, KeyBlock, Trees) ->
     int_create_block(BlockHash, Block, KeyBlock, Trees, Txs).
 
 int_create_block(PrevBlockHash, PrevBlock, KeyBlock, Trees, Txs) ->
+    %% Micro block always has the same protocol version as the last key block.
+    Protocol = aec_blocks:version(KeyBlock),
     PrevBlockHeight = aec_blocks:height(PrevBlock),
 
-    %% Assert correctness of last block protocol version, as minimum
-    %% sanity check on previous block and state (mainly for potential
-    %% stale state persisted in DB and for development testing).
-    ExpectedPrevBlockVersion =
-        aec_hard_forks:protocol_effective_at_height(PrevBlockHeight),
-    {ExpectedPrevBlockVersion, _} = {aec_blocks:version(PrevBlock),
-                                     {expected, ExpectedPrevBlockVersion}},
-
```

Comment:
Shouldn't this sanity check stay there in some new form?

Reply:
All the blocks are added via the conductor process which makes sure each added block is valid and connected to the genesis. So we can rely on the previous key block to be valid and so having the right protocol version.